src/worker/main.ts

```diff
@@ -1,24 +1,65 @@
 import { readExcel } from "./read-wordlist";
-import { SaveFileEventPayload } from "@common/events";
+import {
+  EventPayload,
+  SaveFileRequestPayload,
+  CompileModelRequestPayload,
+  CompileModelResponsePayload,
+  EventType,
+} from "@common/events";
+import {
+  compileModelFromLexicalModelSource,
+  WordListFromArray,
+} from "@predictive-text-studio/lexical-model-compiler";
 import Storage from "./storage";
 
 const storage = new Storage();
 
-// Print the function's name (even though that's self-evident) just to get the
-// linter and Rollup to acknowledge that readExcelSync() is used.
-self.postMessage(`hello, from worker! I can use ${readExcel.name}`);
-
-const handleSaveFileEvent = async (event: MessageEvent) => {
-  const payload = event.data as SaveFileEventPayload;
+const handleEvent = async (event: MessageEvent) => {
+  const payload = event.data as EventPayload;
+  switch (payload.type) {
+    case EventType.SaveFileRequest: {
+      await handleSaveFileEvent(payload as SaveFileRequestPayload);
+    }
+    case EventType.CompileModelRequest: {
+      await handleCompileModel(payload as CompileModelRequestPayload);
+    }
+  }
+};
 
+const handleSaveFileEvent = async (payload: SaveFileRequestPayload) => {
   try {
     await storage.saveFile(payload.name, await payload.file.arrayBuffer());
+    const response: EventPayload = { type: EventType.SaveFileResponse };
+    postMessage(response);
   } catch (e) {
-    postMessage("Save Failed");
-    return;
+    const response: EventPayload = {
+      type: EventType.SaveFileResponse,
+      errorMessage: "Save Failed",
+    };
+    postMessage(response);
   }
+};
```

Comment:
This `try/catch` should not attempt catching the error from `postMessage()`. So use one `postMessage()` after the `try/catch`.

Reply:
> Thank you for taking time to do this! However, I'm honestly not sure about this :/
> 
> It kind of touches upon #74.
> 
> So we could accept the changes here, and then focus _exclusively_ on refactoring and tests for the few PRs.
> 
> I would also like to see tests for this. Does this actually... you know... work? 
> 
> So how would you like to continue @jienius? Would you like to get this merged and then focus our attention next to testing, documenting, and refactoring? Or would you rather get that over with right now?
> 
> I feel like we're dragging on our "proof-of-concept" period, and I'd like to see more "production-ready" code. This means tests, documentation, and better factoring!

I will try to address your comments and add more tests. I don't expect this PR to be merged any time soon tbh, there are just so many... uncertainty.